File: frontend/src/types/lesson.ts
export interface Lesson {
  id: number
  chapterId: number
  title: string
  lessonType: number
  videoUrl: string | null
  documentUrl: string | null
  duration: number | null
  content: string | null
  orderIndex: number
}

export interface CreateLessonPayload {
  title: string
  lessonType: number
  videoUrl?: string
  documentUrl?: string
  duration?: number
  content?: string
  orderIndex: number
}

export interface UpdateLessonPayload extends CreateLessonPayload {}

export interface ReorderLessonsPayload {
  lessonIds: number[]
}

export const LESSON_TYPE_LABEL: Record<number, string> = {
  0: 'Video',
  1: 'Tài liệu',
  2: 'Bài đọc',
}

export const LESSON_TYPE_OPTIONS = [
  { value: 0, label: 'Video' },
  { value: 1, label: 'Tài liệu' },
  { value: 2, label: 'Bài đọc' },
]

File: backend/src/main/java/com/weblearning/entity/Lesson.java
package com.weblearning.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "lessons")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Lesson {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chapter_id", nullable = false)
    private Chapter chapter;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(length = 180)
    private String slug;

    @Column(nullable = false)
    private Integer lessonType;

    @Column(length = 500)
    private String videoUrl;

    @Column(length = 500)
    private String documentUrl;

    @Column
    private Integer duration;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false)
    private Integer orderIndex;

    @Column(nullable = false)
    private boolean deleted;

    @Column
    private LocalDateTime deletedAt;
}
